### PR TITLE
Clean up tools and js-envs.

### PR DESCRIPTION
**Pull request by sjrd**
* Remove deprecated stuff
* Make as many things as possible private

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @gzm0 

Pay particular attention to the privatizations in `jsenv.phantomjs`. I wasn't completely sure I could privatize the things I made private in there.

File: js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala [hunk 1]
```diff
@@ -66,7 +66,31 @@ class PhantomJSEnv(
...
+
+    private var mgrIsRunning: Boolean = false
+
+    private object websocketListener extends WebsocketListener { // scalastyle:ignore
+      def onRunning(): Unit = ComPhantomRunner.this.synchronized {
+        mgrIsRunning = true
+        ComPhantomRunner.this.notifyAll()
+      }
+
+      def onOpen(): Unit = ComPhantomRunner.this.synchronized{
```
Comment (gzm0) [inline js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala]: Space before {

File: js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala [hunk 2]
```diff
@@ -66,7 +66,31 @@ class PhantomJSEnv(
...
+        ComPhantomRunner.this.notifyAll()
+      }
+
+      def onClose(): Unit = ComPhantomRunner.this.synchronized{
```
Comment (gzm0) [inline js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala]: Space before {

**Comment by gzm0** [comment]
That's all.
I didn't check if there is more things we could make private. If you want me to, I'll need some more time.

**Comment by sjrd** [comment]
Updated (just fixed the two missing spaces).

> I didn't check if there is more things we could make private. If you want me to, I'll need some more time.

No, that's fine. I used a very methodical approach, systematically going through all files.

**Comment by gzm0** [comment]
LGTM

**Pull request merged**